### ISSUE TITLE
Discussion: Limiting line width in the source.

### ISSUE THREAD

**Issue by dpriskorn** [Dolibarr/dolibarr#14407]
# Discussion
Is there a policy on line width for Dolibarr? I see very long lines in some files and its hard to read. I suggest limiting to column 79 max and add that to the linter or whatever checks the code.

WDYT?

related to: https://github.com/Dolibarr/dolibarr/issues/10054

**Comment by eldy** [comment]
Limit of 79 char is coming for the 1980 years when terminal were text terminal with a limit of 80 characters. It is often not enough for just a simple error message and there is no reason to take several line for a hard coded technical error message.
Today, there is no reason to keep a limit of 80.
Recommended is 120 (as defined in https://www.php-fig.org/psr/psr-12/)
Returning warnings/errors in CI is 500

**Comment by dpriskorn** [comment]
I respectfully disagree. I code on a small horizontal 14" laptop screen and I find it very useful to have 2 windows side by side in emacs.
I use glasses because I'm nearsighted and its hard to read small font on the screen.

Wih 79 char lines this is not an issue.  
When reading the Dolibarr source I have to move my eyes a lot and some times scroll left and right because of varying line length. Line wrapping is not a solution because it makes it really hard to understand the code.
Its also really hard to view the code on github where you have to scroll sideways a lot.
On mobile phones its even harder because of the smaller screen. I use octodroid to edit issues and its usefull to jump to the code, but I can't read it in octodroid because of all the long lines.

See PEP8 https://www.python.org/dev/peps/pep-0008/#maximum-line-length where IMO good arguments are given.
See also https://www.emacswiki.org/emacs/EightyColumnRule
And https://www.kernel.org/doc/Documentation/process/coding-style.rst

**Comment by egils-consulting** [comment]
Thats not compatible with the current roadmap. Also it coild lead to up to 6 supported versions which is unacceptable IMO.
No dates. People can update when they run into trouble or every year. Nothing keeping users from setting a calendar up to remind them every year.
I updated from 9->12 a few weeks ago and it took 20 min. using git plus testing time of the new version.

**Comment by dpriskorn** [comment]
Here is an example screenshot of 2 windows in emacs in my terminal with a bigger fontsize I use sometimes. On the left a random php-file from Dolibarr and on the right a 79 char max python script I wrote recently. Line wrapping is enabled.
Do you see the difference?
![Skärmbild_2020-08-07_18-38-34](https://user-images.githubusercontent.com/68460690/89673842-6282b680-d8e7-11ea-90a4-65e6bd3d8f3e.png)

**Comment by dpriskorn** [comment]
Here is without line wrapping and a little further down in the python script:
![Skärmbild_2020-08-07_19-55-30](https://user-images.githubusercontent.com/68460690/89674179-08362580-d8e8-11ea-9010-452121f06661.png)

**Comment by dpriskorn** [comment]
Here is without line wrapping and only the Dolibarr file:
![Skärmbild_2020-08-07_19-56-32](https://user-images.githubusercontent.com/68460690/89674265-2865e480-d8e8-11ea-8440-f3e960d96f4f.png)

**Comment by dpriskorn** [comment]
@eldy How does your setup look like? I imagine you have a wider screen than 14" and a small font?

**Comment by dpriskorn** [comment]
I calculated the lines longer than 79 chars in the file above to 16%.  https://github.com/Dolibarr/dolibarr/blob/develop/htdocs/commande/card.php
I think this is representative for the whole project. From what I understood readability affects maintainability. Uglier/complex cumbersome code leads to fewer eyeballs reading and understanding it which leads to less participation.

**Comment by dpriskorn** [comment]
What about you @ksar-ksar WDYT?

**Comment by eldy** [comment]
Using 2 open files in one screen does not look to be a common practice to code. Why don't use tabs ? I used to edit several files in parallel but it's first time i see 2 files open on same screen for coding or read code :-) Too difficult for me, I prefer to switch using tabs or using the auto swith of modern IDE when clicking on function or method name.

**Comment by ksar-ksar** [comment]
> What about you @ksar-ksar WDYT?

Hi,

I'm not part of the core development team so my thinking is not so important.
But I personally agree with @eldy 80 char is far too low.
I would say that for me less than 200 char is good.

**Comment by dpriskorn** [comment]
Over 300 chars does not seem to be uncommon in the codebase. 
Could we have 200 as a limit?
I can also jump easily between functions and files in emacs but scrolling horizontally and zooming in/out is a pain.
I can use another GUI editor but I'm quite happy with the text editing and debugging qualities of emacs.
I have actually never seen or contributed to a project with as long lines as Dolibarr, but I mostly code in python and rust.
